src/routers/alpha-router/quoters/base-quoter.ts

```diff
@@ -107,19 +114,41 @@ export abstract class BaseQuoter<Route extends V2Route | V3Route | MixedRoute> {
     gasModel?: IGasModel<RouteWithValidQuote>,
     gasPriceWei?: BigNumber
   ): Promise<GetQuotesResult> {
+    const beforeGetRoutesThenQuotes = Date.now();
+
     return this.getRoutes(tokenIn, tokenOut, tradeType, routingConfig)
-      .then((routesResult) =>
-        this.getQuotes(
-          routesResult.routes,
-          amounts,
-          percents,
-          quoteToken,
-          tradeType,
-          routingConfig,
-          routesResult.candidatePools,
-          gasModel,
-          gasPriceWei
-        )
+      .then((routesResult) => {
+          const beforeGetQuotes = Date.now();
+
+          return this.getQuotes(
+            routesResult.routes,
+            amounts,
+            percents,
+            quoteToken,
+            tradeType,
+            routingConfig,
+            routesResult.candidatePools,
+            gasModel,
+            gasPriceWei
+          ).then((quotesResult) => {
+            metric.putMetric(
+              `${this.quoterVersion}GetQuotesLoad`,
```

Comment:
do we know that this metric will not collide with the other existing metric of similar name?

Reply:
Good point. Although the existing metric is called `*QuotesLoad`, it might be confusing when we add this one. I think we can name it something like `*OverallGetQuotesLoad`, wdyt?